### PR TITLE
Object ID validation - Custom Fields

### PR DESCRIPTION
**Pull request by kpbacode**
https://github.com/killbill/killbill-admin-ui/pull/317

### CODE REVIEW

**Comment by kpbacode** [comment]
Removed gon. Hardcoded URL for to check if tests will pass.

**Comment by kpbacode** [comment]
Invoice Items search added.

To do:
- Tests

**Comment by kpbacode** [comment]
will add more tests later today

**Comment by kpbacode** [comment]
Test done. Ready for review and merge if all good. thanks

File: config/locales/en.yml [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+  transfer: "Transfer"
+  comment: "Comment"
   unknown_bcd: "Unknown bill cycle day"
   payment_method_id: "Payment Method ID:"
   external_key_with_colon: "External Key:"
   external_id_with_colon: "External ID:"
   name_with_colon: "Name:"
+  custom_field_uuid_exist_in_account_db: "UUID do exist in ACCOUNT object database."
+  custom_field_uuid_exist_in_bundle_db: "UUID do exist in BUNDLE object database."
+  custom_field_uuid_exist_in_subscripction_db: "UUID do exist in SUBSCRIPCTION object database."
```
Comment (pierre) [inline config/locales/en.yml:13]: Typo: `SUBSCRIPTION`

File: app/controllers/kaui/custom_fields_controller.rb [hunk 2]
```diff
@@ -39,8 +43,93 @@ def pagination
...
+
+    param_uuid = params[:uuid]
+
+    begin
+      test_uuid = Kaui::Invoice.find_by_invoice_item_id(param_uuid,false,'NONE', options_for_klient)
+
+    rescue StandardError
+    ensure
+      if !test_uuid.blank?
+        msg = { status: '200', message: I18n.translate('custom_field_uuid_exist_in_invoice_item_db')  }
```
Comment (pierre) [inline app/controllers/kaui/custom_fields_controller.rb:59]: Few thoughts:

1. I don't think we should display anything if the object (e.g. invoice item) exists, just a warning if it doesn't
2. If the target object for the custom field doesn't exist, we should prevent the user from being able to save it
3. Can we send the object type to the back-end? That way, we don't have to try all endpoints (invoice item, bundle, etc.). That might take a while 😅

Comment (kpbacode) [reply app/controllers/kaui/custom_fields_controller.rb:59]: Yes, good points. Got it.

Comment (kpbacode) [reply app/controllers/kaui/custom_fields_controller.rb]: 1. Done.
2. Pending.  
3. Done. Way better. 
There will be some improvements. 
https://github.com/killbill/killbill-admin-ui/pull/319/commits/11b43ff5f25250dcd38f94e9eb17d046f2ab56ef

**Comment by kpbacode** [comment]
To do:

1. I18N
2. Tests

File: app/controllers/kaui/custom_fields_controller.rb (hunk 2 above)
Comment (kpbacode) [reply app/controllers/kaui/custom_fields_controller.rb]: 2. Done

Comment (kpbacode) [reply app/controllers/kaui/custom_fields_controller.rb]: 2. Was already done but response was from KillBill API without I18N and raw data.

**Comment by kpbacode** [comment]
Fixed tests
I18N added
When Object ID do not exist and you want to add there is redirection to /custom_fields

**Pull request closed without merging**